Warning squashes

pavanbalaji opened the pull request:
## What
Squashing a bunch of warnings

## Why ?
Because warnings are annoying.  Doh.

## How ?
Using awesome engineering!

swx-jenkins commented:
Can one of the admins verify this patch?

dmitrygladkov commented:
ok to test

swx-jenkins commented:
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/6560/ for details.

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9316/ for details (Mellanox internal link).

dmitrygx committed:
UCT/TCP: Add support for messages with the length > 64k

dmitrygx committed:
UCT/TCP,GTEST: Add test and use uint32_t instead usngined

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9375/ for details (Mellanox internal link).

swx-jenkins commented:
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/6613/ for details.

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9382/ for details (Mellanox internal link).

swx-jenkins commented:
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/6620/ for details.

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9383/ for details (Mellanox internal link).

swx-jenkins commented:
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/6621/ for details.

dmitrygx committed:
UCT/TCP: Fix possible fd leak

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9407/ for details (Mellanox internal link).

swx-jenkins commented:
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/6640/ for details.

swx-jenkins commented:
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/6642/ for details.

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9409/ for details (Mellanox internal link).

swx-jenkins commented:
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/6653/ for details.

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9420/ for details (Mellanox internal link).

swx-jenkins commented:
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/6661/ for details.

pavanbalaji committed:
remove __clear_cache declaration.

It is unclear why this declaration was added in the first place.
Needs to be investigated.

minsii committed:
backtrace_next should defined as static function.

minsii committed:
fix old style inline

pavanbalaji committed:
ad-hoc type casting warnings

This is an ugly patch that does ad-hoc type casting.  The correct
solution is to the refactor the code to not require such casting.

Co-authored-by: Min Si <msi@anl.gov>

minsii committed:
add missing header includes

These are needed for missing declarations of structures or functions.

pavanbalaji committed:
fixes for -Wpointer-arith warnings

Co-authored-by: Min Si <msi@anl.gov>

pavanbalaji committed:
fixes for -Wstrict-prototypes and -Wmissing-prototypes

Co-authored-by: Min Si <msi@anl.gov>

pavanbalaji committed:
fixes for -Wshadow warnings

Some of these names are used for global variables or functions.  Do
not use the same name.

The naming convention here might need to be reviewed.

pavanbalaji committed:
fixes for -Wnested-externs warnings

pavanbalaji committed:
fixes for -Wnested-externs warnings (messy macros version)

pavanbalaji committed:
fixes for -Wtype-limits warnings

pavanbalaji committed:
fixes for -Wundef warnings

Disallow direct #if VAR check.  We need to first check if VAR is
defined before using it.

Co-authored-by: Min Si <msi@anl.gov>

pavanbalaji committed:
workaround platform-specific limitations

pavanbalaji committed:
ad-hoc warning squashes

minsii committed:
fixes for -Wmissing-prototypes using UCS_CLASS_DECLARE

minsii committed:
fixes warnings for glibc endian conversion

IB transport uses the Glibc endian conversion interfaces (e.g., htobe64)
which requires definition of _BSD_SOURCE (for glibc <= 2.19) or
_DEFAULT_SOURCE (for glic > 2.19). Note that infiniband/mlx5dv.h header
also requires these functions. Thus, this patch adds it in CFLAGS at
configure time.

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9437/ for details (Mellanox internal link).

swx-jenkins commented:
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/6663/ for details.

swx-jenkins commented:
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/6664/ for details.

minsii committed:
define declare macro for UCS_PROFILE_FUNC*

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9434/ for details (Mellanox internal link).

swx-jenkins commented:
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/6666/ for details.

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9438/ for details (Mellanox internal link).

swx-jenkins commented:
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/6667/ for details.

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9440/ for details (Mellanox internal link).

mellanox-github commented:
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9441/ for details (Mellanox internal link).

mellanox-github commented:
<details><summary>Mellanox CI: <b>ABORTED</b> on 1 workers (click for details)</summary>

**Note:** the logs will be deleted after 15-Nov-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/61d7c76ed15726cbbbd40625e9d71d7c/raw/18c18f5a6f035fee7eb74194b9e5ee5e33871216/_main) | :question: ABORTED

gleon99 commented:
Hi @pavanbalaji - we're cleaning up the UCX PR backlog and plan to close this PR in 7 days. Please let us know if you'd like to keep it open.
